flink-runtime/src/test/java/org/apache/flink/runtime/minicluster/CountDownLatchedSender.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.minicluster;
+
+import org.apache.flink.runtime.execution.Environment;
+import org.apache.flink.runtime.io.network.api.writer.RecordWriter;
+import org.apache.flink.runtime.jobgraph.tasks.AbstractInvokable;
+import org.apache.flink.types.IntValue;
+
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * A sender that waits for all receivers running.
+ */
+public class CountDownLatchedSender extends AbstractInvokable {
```

Comment:
I would prefer these (CountDowon*(Sender|Retriever)`to either be inner classes in the `MiniClusterITCase` / a new utility classes, or allow users to provide a `Runnable` to execute within `invoke`. They are too tightly coupled to be separate imo.